pyfr/plugins/writer.py

```diff
@@ -58,6 +58,18 @@ def __call__(self, intg):
         else:
             metadata = None
 
+        # Fetch data from other plugins and add it to metadata with ad-hoc keys
+        for csh in intg.completed_step_handlers:
+            try:
+                pdata = csh.serialise(intg)
+                prefix = intg.get_plugin_data_prefix(csh)
+            except AttributeError:
+                pdata = {}
+                prefix = ''
```

Comment:
No need to define prefix, as if pdata is empty the offending block which uses prefix will never execute.

Reply:
fixed in PR242